design/list_command_design.md

```diff
@@ -0,0 +1,275 @@
+# Dockerless CLI - `list` action
+
+This document is the design document for the list CLI command for the Dockerless project. The main functionality for the list CLI command has been implemented. Now all that is left is to implement the following flags:
+
+1. `-i, --inspect`
+2. `-k, --kind`
+3. `-l, --limit`
+5. `-r, --role`
+6. `-s, --search`
+
+In additional to implementing the above flags, we will need to also address the following:
+
+1. Ensure UTs and integration tests exist and are GA quality
+2. Improve error handling experience
+3. Write proper documentation in `README.md` of the repository
+
+```bash
+Name
+----
+  list - Lists all available resources beloging to this account
+
+
+Usage
+----
+  conjur [global options] list [arguments...]
+
+
+Global options
+----
+  -h, --help         Show this help message and exit
+  -d, --debug				 Enable debugging output
+
+       
+Commands
+----
+--[no-]inspect       Show full object information
+
+-k, --kind=arg       Filter by kind (default: none)
+
+-l, --limit=arg      Maximum number of records to return (default: none)
```

Comment:
I just realized that `--limit` isn't really all that useful without a complementary `--offset` flag too

Reply:
For clarity purposes? So if the user has 100 entries they can start in the middle let's say?

Reply:
FYI @InbalZilberman

Reply:
I totally agree with @sgnn7 limit and offset are a pair
Does the REST API support offset?
If not I rather we support what we can and start to write the gaps in an orderly way so we can take them as we go along.

Reply:
Yes the REST API supports offset. Is this back in scope?

Reply:
@sigalsax I think both ways will work but sure lets make it simpler
please notice that in Ruby only these option mention with '=' are you sure there is no technical reason for it?

Reply:
did you mean to comment this on the above comment thread? https://github.com/cyberark/conjur-api-python3/pull/97#discussion_r528612619

Reply:
So yes lets do it in lower priority though